lib/susedistribution.pm

```diff
@@ -854,6 +854,12 @@ sub console_selected {
         return;
     }
     $self->hyperv_console_switch($console, console_nr($console));
+    
+    if (check_screen 'workqueue_lockup') {
+        record_soft_failure 'bsc#1126782';
+        send_key 'ret';
+    }
+
```

Comment:
You are changing a method that is literally used in every test scenario so I strongly suggest to invest heavily in testing . in general, very likely you know this already but for some reason still did not use it: Please use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen` with non-zero timeout to prevent introducing any timing dependant behaviour, to save test execution time as well as state more explicitly from the testers point of view what are the expected alternatives. For example:

```perl
assert_screen([qw(yast2_console-finished yast2_missing_package)]);
if (match_has_tag('yast2_missing_package')) {
    send_key 'alt-o';  # confirm package installation
    assert_screen 'yast2_console-finished';
}
```

Doing an additional explicit `check_screen` in every individual test scenario even though in most cases it is not necessary has a bad performance impact and makes the test results harder to read.

Why not just add this tag to `$args{tags}`? However none of this change should be necessary at all. Just create proper workaround needles with the tags like "root-console" or whatever the test looks for, with a link to the bug and the workaround property set? These needles are always preferred in matches. You only need to apply the workaround if whatever needle that matched has the additional tag you want to use for the workaround, i.e. after the `assert_screen` further below

Reply:
Thanks for you review, I dont think a workaround needle would be sufficient as we need to press enter to regain a working console.
I will add it to the args tags, that seems reasonable

Reply:
Adapted it that way

Reply:
Sorry if I could not make myself clear. Of course you need the workaround action. But you should not need any new tags mentioned in the source code as "match_has_tag" can check for the presence of any tag in the matched needle even though the corresponding "assert_screen" might not have specified it, see https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L479